app/models/spam_report.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+class SpamReport
```

Comment:
I extracted this object out of the mailer because the mailer logic has become quite complex.

I have done this for the following reasons:
- I wanted to test the delegation between Comment and RfC in isolation (without rendering email).
- The email feature would become too complex with this much logic (single responsibility).

Putting this into the model is a bit debatable. I think in smaller Rails this is a valid strategy.

Another approach could have been to create separate mailers for comments and RfCs. I decided against this because I think this approach keeps more of the business logic out of the mailer.